src/main/java/com/teragrep/pth_06/planner/HdfsDBClient.java

```diff
@@ -62,19 +62,23 @@ public class HdfsDBClient {
     private final Logger LOGGER = LoggerFactory.getLogger(HdfsDBClient.class);
     private final String path;
     private final FileSystem fs;
-    private final String topicsRegexString;
+    private final TopicFilter topicFilter;
     private final long ignoreBeforeEpoch;
 
     public HdfsDBClient(Config config, String topicsRegexString) throws IOException {
-        this.topicsRegexString = topicsRegexString;
+        this(config, topicsRegexString, new FileSystemFactoryImpl(config.hdfsConfig).fileSystem(true));
+    }
+
+    public HdfsDBClient(Config config, String topicsRegexString, FileSystem fs) {
+        this.topicFilter = new TopicFilter(topicsRegexString);
```

Comment:
`new TopicFilter(...);` should be in secondary constructor

Reply:
Solved in commit 36647079098ca349c78f381071c5eeccabd02538